persistence/jdbc/src/main/java/org/infinispan/persistence/jdbc/configuration/AbstractJdbcStoreConfigurationSerializer.java

```diff
@@ -0,0 +1,83 @@
+package org.infinispan.persistence.jdbc.configuration;
+
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.XMLStreamWriter;
+
+import org.infinispan.commons.configuration.attributes.AttributeDefinition;
+import org.infinispan.commons.configuration.attributes.AttributeSet;
+import org.infinispan.configuration.serializing.AbstractStoreSerializer;
+import org.infinispan.configuration.serializing.XMLExtendedStreamWriter;
+import org.infinispan.persistence.jdbc.connectionfactory.PooledConnectionFactory;
+
+import static org.infinispan.configuration.serializing.SerializeUtils.writeOptional;
+
+/**
+ * AbstractJdbcStoreConfigurationSerializer.
+ *
+ * @author Tristan Tarrant
+ * @since 9.0
+ */
+public abstract class AbstractJdbcStoreConfigurationSerializer extends AbstractStoreSerializer {
+   protected void writeJdbcStoreAttributes(XMLExtendedStreamWriter writer, AbstractJdbcStoreConfiguration configuration) throws XMLStreamException {
+      configuration.attributes().write(writer);
+   }
+
+   private void writeJDBCStoreConnection(XMLExtendedStreamWriter writer, SimpleConnectionFactoryConfiguration configuration) throws XMLStreamException {
+      writer.writeStartElement(Element.SIMPLE_CONNECTION);
+      writeOptional(writer, Attribute.CONNECTION_URL, configuration.connectionUrl());
+      writeOptional(writer, Attribute.DRIVER_CLASS, configuration.driverClass());
+      writeOptional(writer, Attribute.USERNAME, configuration.username());
+      writeOptional(writer, Attribute.PASSWORD, configuration.password());
+      writer.writeEndElement();
+   }
+
+   private void writeJDBCStoreConnection(XMLExtendedStreamWriter writer, PooledConnectionFactoryConfiguration configuration) throws XMLStreamException {
+      writer.writeStartElement(Element.CONNECTION_POOL);
+      writeOptional(writer, Attribute.CONNECTION_URL, configuration.connectionUrl());
+      writeOptional(writer, Attribute.DRIVER_CLASS, configuration.driverClass());
+      writeOptional(writer, Attribute.USERNAME, configuration.username());
+      writeOptional(writer, Attribute.PASSWORD, configuration.password());
+      writer.writeEndElement();
+   }
+
+   private void writeJDBCStoreConnection(XMLExtendedStreamWriter writer, ManagedConnectionFactoryConfiguration configuration) throws XMLStreamException {
+      writer.writeStartElement(Element.DATA_SOURCE);
+      writer.writeAttribute(Attribute.JNDI_URL, configuration.jndiUrl());
+      writer.writeEndElement();
+   }
+
+   protected void writeJDBCStoreConnection(XMLExtendedStreamWriter writer, AbstractJdbcStoreConfiguration configuration) throws XMLStreamException {
```

Comment:
Only the binary jdbc serializes its connection. Shouldn't the string-based and mixed do it too?